pyccel/ast/operators.py

```diff
@@ -115,8 +113,8 @@ def __init__(self, *args):
         if self.stage == 'syntactic':
             super().__init__()
             return
-        self._set_dtype()
-        self._set_shape_rank()
+        self._dtype, self._precision = self._calculate_dtype(*self._args)
```

Comment:
I think this pylint error can be deactivated with a comment. `PyccelOperator` doesn't have this function as it is an abstract class. The only other way of doing this would be to define the functions but have them do nothing but raise an error

Reply:
fixed in [f09813c](https://github.com/pyccel/pyccel/pull/797/commits/f09813cc5a2ea6be686a649e8feeb2ec1abab23a)

Reply:
The error should only be deactivated at this line, not for the whole file